legend-engine-xt-javaGeneration-pure/src/main/resources/core_external_language_java/metamodel_factories.pure

```diff
@@ -1453,6 +1453,12 @@ function meta::external::language::java::factory::j_cast(expression:Code[1], cas
    if($castTo == javaDouble() && $expression.type->isBoxedNumberType(),
      | ^CastProxy(expression=$expression, proxy=$expression->j_invoke('doubleValue', [], javaDouble()), type=$castTo),
      |
+   if($castTo == javaBigDecimal() && $expression.type==javaDouble(),
+     | ^CastProxy(expression=$expression, proxy=javaBigDecimal()->j_invoke('valueOf', $expression, $castTo), type=$castTo),
+     |  
+   if($castTo == javaBigDecimal() && $expression.type==javaLong(),
+     | ^CastProxy(expression=$expression, proxy=javaBigDecimal()->j_invoke('valueOf', $expression, $castTo), type=$castTo),
+     |    
```

Comment:
It is true.  Assumed the cast gets applied if you put in a Long and the expected type is Decimal.   Quite easy to add j_bigDecimal() utility, it depends on what is natural to write.  Let me know - happy to add the utils instead of cast

Reply:
Removed the cast capability.